pycbc/events/significance.py

```diff
@@ -0,0 +1,253 @@
+    # normalised to 1
+    bg_above = back_stat > fit_thresh
+    bg_above_thresh = np.sum(dec_facs[bg_above])
+    fg_above = fore_stat > fit_thresh
+
+    # These will be overwritten, but just to silence a warning
+    # in the case where trstats.cum_fit returns zero
+    back_cnum = np.zeros_like(back_stat)
+    fnlouder = np.zeros_like(fore_stat)
+
+    # Ue the fit above the threshold
+    back_cnum[bg_above] = trstats.cum_fit(fit_func, back_stat[bg_above],
+                                          alpha, fit_thresh) * bg_above_thresh
+    fnlouder[fg_above] = trstats.cum_fit(fit_func, fore_stat[fg_above],
+                                         alpha, fit_thresh) * bg_above_thresh
+
+    # below the fit threshold, we count the number of louder events,
+    # as things get complicated by clustering below this point
+    fg_below = np.logical_not(fg_above)
+    bg_below = np.logical_not(bg_above)
+
+    back_cnum[bg_below], fnlouder[fg_below] = \
+        n_louder(back_stat, fore_stat, dec_facs)
+
+    return back_cnum, fnlouder
+
+
+_significance_meth_dict = {
+    'trigger_fit': trig_fit,
+    'n_louder': n_louder
+}
+
+
+def calculate_n_louder(method_dict, back_stat, fore_stat, dec_facs):
+    """
+    Wrapper to find the correct n_louder calculation method using standard
+    inputs
+    """
+    calculation_method = method_dict['method']
+    fit_func = method_dict['function']
+    fit_thresh = method_dict['threshold']
+    return _significance_meth_dict[calculation_method](back_stat, fore_stat,
+                                                       dec_facs,
+                                                       fit_func=fit_func,
+                                                       fit_thresh=fit_thresh)
+
+
+def insert_significance_option_group(parser):
+    """
+    Add some options for use when a significance is being estimated from
+    events or event distributions.
+    """
+    parser.add_argument('--far-calculation-method', nargs='+',
+                        default=[],
+                        help="Method used for FAR calculation in each "
+                             "detector combination, given as "
+                             "combination:method pairs, i.e. "
+                             "H1:trigger_fit H1L1:n_louder H1L1V1:n_louder "
+                             "etc. Method options are ["
+                             + ",".join(_significance_meth_dict.keys()) +
+                             "]. Default = n_louder for all not given")
+    parser.add_argument('--fit-threshold', nargs='+', default=[],
+                        help="Thresholds for the fits to statistic "
+                             "values for FAN approximation if "
+                             "--far-calculation-method is 'trigger_fit'. "
+                             "Given as combination-values pairs, e.g. "
+                             "H1:0 L1:0 V1:-4, for all combinations which "
```

Comment:
This is a bit verbose, basically you are saying 'if far-calculation-method=trigger_fit' twice in the same message. 
Suggest 'Trigger statistic fit thresholds for FAN estimation, given as combination-value pairs ex. H1:0 L1:0 V1:-4 for all combinations with --far-calculation-method = trigger_fit'